query/common/src/test/groovy/org/codice/ddf/admin/common/fields/base/BaseFunctionFieldTest.groovy

```diff
@@ -0,0 +1,134 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.common.fields.base
+
+import org.codice.ddf.admin.api.DataType
+import org.codice.ddf.admin.api.fields.FunctionField
+import org.codice.ddf.admin.common.fields.base.scalar.StringField
+import org.codice.ddf.admin.common.report.message.DefaultMessages
+import org.codice.ddf.admin.common.report.message.ErrorMessage
+import spock.lang.Specification
+
+class BaseFunctionFieldTest extends Specification {
+
+    BaseFunctionField functionField
+
+    def 'Argument paths are updated with function field name is updated'() {
+        setup:
+        functionField = new TestBaseFunctionField(true)
+
+        expect:
+        functionField.path() == [TestBaseFunctionField.DEFAULT_NAME]
+        functionField.getArguments()[0].path() == [TestBaseFunctionField.DEFAULT_NAME, BaseFunctionField.ARGUMENT, StringField.DEFAULT_FIELD_NAME]
+
+        when:
+        functionField.fieldName('newName')
+
+        then:
+        functionField.path() == ['newName']
+        functionField.getArguments()[0].path() == ['newName', BaseFunctionField.ARGUMENT, StringField.DEFAULT_FIELD_NAME]
+    }
+
+    def 'Function field validates arguments'() {
+        setup:
+        functionField = new TestBaseFunctionField(true)
+
+        when:
+        def report = functionField.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages()[0].getCode() == DefaultMessages.EMPTY_FIELD
+        report.messages()[0].getPath() == [TestBaseFunctionField.DEFAULT_NAME, BaseFunctionField.ARGUMENT, StringField.DEFAULT_FIELD_NAME]
+    }
+
+    def 'Adding result message has correct path on added message'() {
+        setup:
+        functionField = new TestBaseFunctionField(false)
+        functionField.addResultMessage(new ErrorMessage('test'))
+
+        when:
+        def report = functionField.getValue()
+
+        then:
+        report.messages().size() == 1
+        report.messages()[0].getPath() == [TestBaseFunctionField.DEFAULT_NAME]
+    }
+
+    def 'Setting function arguments'() {
+        setup:
+        functionField = new TestBaseFunctionField(false)
+
+        expect:
+        functionField.getArguments()[0].getValue() == TestBaseFunctionField.ARG_VALUE
+
+        when:
+        def value = [(StringField.DEFAULT_FIELD_NAME): 'test' ]
+        functionField.setValue(value)
+
+        then:
+        functionField.getArguments()[0].getValue() == 'test'
+    }
+
+    def 'Updating path updates arguments paths'() {
+        setup:
+        functionField = new TestBaseFunctionField(false)
+
+        when:
+        functionField.updatePath(['test'])
+
+        then:
+        functionField.path() == ['test', TestBaseFunctionField.DEFAULT_NAME]
+        functionField.getArguments()[0].path() == ['test', TestBaseFunctionField.DEFAULT_NAME, FunctionField.ARGUMENT, StringField.DEFAULT_FIELD_NAME]
+    }
+
+    class TestBaseFunctionField extends BaseFunctionField<StringField> {
```

Comment:
Is there a reason we don't reuse the functions in the `TestFieldProvider` class?